### ISSUE TITLE
Can't use DownloadBeforeOpen file behavior offline

### ISSUE THREAD

**Issue by devsales** [realm/realm-js#5986]
### How frequently does the bug occur?

Always

### Description

```JavaScript
const realmFileBehavior = {
  type: OpenRealmBehaviorType.DownloadBeforeOpen,
  timeOut: 1000,
  timeOutBehavior: "openLocalRealm", // open the local realm if the download has not completed within 1000ms
};
```
Using this realm file configuration does not work when the user is offline. When the app is launched the RealmProvider is stuck on the fallback component, e.g. a loading screen, and the user is not authenticated. So for some reason the time out does not work.

When using `OpenRealmBehaviorType.OpenImmediately` it works fine. See also [Issue 5385](https://github.com/realm/realm-js/issues/5385).

### Can you reproduce the bug?

Always

### Reproduction Steps

Use this existing file behavior config:
```JavaScript
const realmFileBehavior = {
  type: OpenRealmBehaviorType.DownloadBeforeOpen,
  timeOut: 1000,
  timeOutBehavior: "openLocalRealm", // open the local realm if the download has not completed within 1000ms
};
```
Run app in flight mode. I can reproduce this only on a real device.

### Version

"@realm/react": "0.4.3","realm": "11.8.0"

### What services are you using?

Both Atlas Device Sync and Atlas App Services

### Are you using encryption?

-- select --

### Platform OS and version(s)

MacOS Ventura

### Build environment

Which debugger for React Native: ..

**Comment by kneth** [comment]
@devsales Thank you for reporting. By "user is offline", I assume you mean that the device has lost connection (you mention flight mode).

The fix might be https://github.com/realm/realm-js/pull/5916 but it isn't released yet in any v11 release. I believe the missing enum is fixed and release in our v12 series.

**Comment by devsales** [comment]
@kneth thanks for the reference to #5916. This lead me to go through the code of lib/extensions.js. I saw that in the `open(config)` function, after resolving the `openPromises` the initial subscriptions could be rerun on open with the `rerunOnOpen` flag. In my case the flag was set to true, which I'm guessing caused the issue.

After I set the flag to `false` it worked fine. :)

So if someone wants to use this configuration, they have to make sure that `rerunOnOpen` is not set to `true`.
```JavaScript
const realmFileBehavior = {
  type: OpenRealmBehaviorType.DownloadBeforeOpen,
  timeOut: 1000,
  timeOutBehavior: "openLocalRealm", // open the local realm if the download has not completed within 1000ms
};
```
